doc/lsst.ip.diffim/code_notes.rst

```diff
@@ -0,0 +1,183 @@
+the `hotpants <https://github.com/acbecker/hotpants>`_ C package
+by Becker, while the ZOGY algorithm is entirely implemented in
+Pyton based on mostly on the ``numpy`` FFT functionality.
+
+Beside the top level configuration options, the AL algorithm has its
+own set of separate configuration parameters, while the ZOGY algorithm
+does not have any algorithm specific options.
+
+Alard-Lupton algorithm 
+----------------------
+
+The Alard-Lupton numerical fitting for the coefficients (of the
+composing basis kernel functions) of the image convolution kernel is
+performed independently on a number of stamp images around selected
+sources. These independent solutions are called *kernel candidates*
+throughout the code. The final kernel solution model for the whole
+image is a smoothly varying spatial fit of the coefficients as a
+function of the image pixel coordinates.
+
+If ``convolveTemplate==False`` the science exposure is convolved and
+then the template is subtracted from the science image.
+
+The performance of the AL algorithm was studied in details in the
+[Becker_LDM-227]_ report. This study forms the basis of the AL
+algorithm default values; the the degree of the polynomial
+multiplicator of the Gaussian kernel basis functions (``degGauss``),
+the degree of the polynomial that is fitted to the spatial variaton of
+the solution coefficients accross the image (``spatialKernelOrder``)
+and the default detection thresholds (5.5 sigma). As a legacy of this
+study, the ``Winter2013ImageDifferenceTask`` is still available in
+``imageDifference.py`` though it is unclear which test data repository
+it requires.
+
+Due to noise in the template image, convolving the template introduces
+correlation in the noise in the template image. The AL algorithm was
+improved by an additional *afterburner* decorrelation to remove the
+noise correlation in the image difference. The implemented
+decorrelation method and its mathematical formulae of the
+decorrelation kernel is summarised and studied in [Reiss_DMTN-021]_.
+
+Zackay-Ofek-Gal-Yam algorithm
+-----------------------------
+
+[ZOGY2016]_ is free from the assumption that the template is noise
+free or specially selected by any other means. We simply deal with two
+images with different PSFs and noise characteristics (sigma). In the
+basic version of the algorithm, the random noise in the pixels are
+assumed to be background dominated i.e. uncorrelated between pixels
+and independent of the pixel values. Also we assume that the noise has
+zero expectation value i.e. the expectation value of the random noise
+is already removed. ZOGY shows that if these assumptions hold, the
+difference image noise is also independent and identically distributed
+over its pixels (white) by construction, there is no need to
+decorrelate the noise in the difference image.
+
+With the exposure variance planes, exactly the same steps are repeated than on
+the data planes, only the subtraction step is replaced by addition.
```

Comment:
Can you please elaborate on this sentence? I don't understand why addition is used instead of subtraction.

Reply:
I think this was to follow the variance addition formula of the difference of uncorrelated random variables in the code. Thinking a bit more about it, this might not be even fully appropriate here, I have to re-check algorithm assumptions in the paper. Commented.

Reply:
OK, I trust you on this one!